### PR TITLE
Update threads example

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update threads example and iota.rs to include max inputs check fix

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

With the example

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: examples/threads.rs [hunk 1]
```diff
@@ -48,36 +50,49 @@ async fn main() -> Result<()> {
...
-                                Address::try_from_bech32(
-                                    "rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu",
-                                )?
-                                .1,
-                            )))
-                            .finish_output()?,
+                            .add_unlock_condition(UnlockCondition::Address(AddressUnlockCondition::new(address_)))
+                            .finish_output()?;
+                        // amount of outputs in the transaction (one additional output might be added for the remaining amount)
+                        1
```
Comment (thibault-martinez) [inline examples/threads.rs:78]: Any reason we explicit this ?

Comment (Thoralf-M) [reply examples/threads.rs:78]: So it's easy to change to another number, like 100 for example

Comment (Thoralf-M) [reply examples/threads.rs:78]: I'm using it for testing like this

Comment (thibault-martinez) [reply examples/threads.rs:78]: Fair enough, was curious

**Review by thibault-martinez — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**